gooddata-flight-server/gooddata_flight_server/config/config.py

```diff
@@ -0,0 +1,515 @@
+    task_close_threads: int
+    task_result_ttl_sec: int
+
+    metrics_host: Optional[str]
+    metrics_port: int
+
+    health_check_host: Optional[str]
+    health_check_port: int
+
+    malloc_trim_interval_sec: int
+    log_event_key_name: str
+    log_trace_keys: Dict[str, str]
+
+    otel_config: OtelConfig
+
+    def without_tls(self) -> "ServerConfig":
+        def _basic_sanity(val: bytes) -> bytes:
+            return val[0:38] + b"..." + val[-38:]
+
+        sanitized_root_cert: Optional[bytes] = None
+        sanitized_cert_and_key: Optional[Tuple[bytes, bytes]] = None
+
+        if self.tls_root_cert is not None:
+            sanitized_root_cert = _basic_sanity(self.tls_root_cert)
+
+        if self.tls_cert_and_key is not None:
+            sanitized_cert_and_key = (
+                _basic_sanity(self.tls_cert_and_key[0]),
+                _basic_sanity(self.tls_cert_and_key[1]),
+            )
+
+        return dataclasses.replace(
+            self,
+            tls_cert_and_key=sanitized_cert_and_key,
+            tls_root_cert=sanitized_root_cert,
+        )
+
+
+class _Settings:
+    ListenHost = "listen_host"
+    ListenPort = "listen_port"
+    AdvertiseHost = "advertise_host"
+    AdvertisePort = "advertise_port"
+    UseTls = "use_tls"
+    UseMtls = "use_mtls"
+    TlsCertificate = "tls_certificate"
+    TlsPrivateKey = "tls_private_key"
+    TlsRoot = "tls_root_certificate"
+    AuthenticationMethod = "authentication_method"
+    TokenHeaderName = "token_header_name"
+    TokenVerification = "token_verification"
+    TaskThreads = "task_threads"
+    TaskCloseThreads = "task_close_threads"
+    TaskResultTtlSec = "task_result_ttl_sec"
+    MetricsHost = "metrics_host"
+    MetricsPort = "metrics_port"
+    HealthcheckHost = "health_check_host"
+    HealthcheckPort = "health_check_port"
+    MallocTrimIntervalSec = "malloc_trim_interval_sec"
+    LogEventKeyName = "log_event_key_name"
+    LogTraceKeys = "log_trace_keys"
+    OtelExporterType = "otel_exporter_type"
+    OtelServiceName = "otel_service_name"
+    OtelServiceNamespace = "otel_service_namespace"
+    OtelServiceInstanceId = "otel_service_instance_id"
+
+
+_LOCALHOST = "127.0.0.1"
+_DEFAULT_ADVERTISE_HOST = socket.gethostbyaddr("127.0.0.1")[0] if platform.system() == "Darwin" else socket.getfqdn()
+_DEFAULT_LISTEN_PORT = 17001
+_DEFAULT_TASK_THREADS = 32
+_DEFAULT_TASK_CLOSE_THREADS = 2
+_DEFAULT_TASK_RESULT_TTL_SEC = 60
+_DEFAULT_MALLOC_TRIM_INTERVAL_SEC = 30
+_DEFAULT_METRICS_PORT = 17101
+_DEFAULT_HEALTHCHECK_PORT = 8877
+_DEFAULT_LOG_EVENT_KEY_NAME = "event"
+_DEFAULT_TOKEN_VERIFICATION = "EnumeratedTokenVerification"
+
+_SUPPORTED_EXPORTERS = [
+    "none",
+    OtelExporterType.Zipkin.value,
+    OtelExporterType.OtlpHttp.value,
+    OtelExporterType.OtlpGrpc.value,
+    OtelExporterType.Console.value,
+]
+
+_SUPPORTED_AUTH_METHOD = [
+    AuthenticationMethod.NoAuth.value,
+    AuthenticationMethod.Token.value,
+]
+
+
+def _s(name: str) -> str:
```

Comment:
Is this function name correct?

Reply:
yep, it's intentionally short.. it is used internally & repetetivly when constructing dynaconf validators. it wraps around the setting name, adds the section.

Reply:
i have renamed it to `_fqsn` -> fully qualified setting name + added doc